MetaMask/metamask-extension
# migration of button icon to use TS box version

## Explanation
Used the new TS box Version in ButtonIcon component 

* Fixes #20161

## Screenshots/Screencaps



### Before


https://github.com/MetaMask/metamask-extension/assets/8112138/ec341e39-e2d8-4246-8309-67446773c3d4



### After

https://github.com/MetaMask/metamask-extension/assets/8112138/bd6ae3fc-c695-49d9-af71-9ba740866723

## Manual Testing Steps

- Go to the storybook build on this branch
- Search `ButtonIcon` 
- Check docs and stories work and make sense

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
